units/src/parse.rs

```diff
@@ -19,19 +20,31 @@ use crate::prelude::*;
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
 pub struct ParseIntError {
-    pub(crate) input: String,
-    // for displaying - see Display impl with nice error message below
+    /// The input string that caused the error.
+    input: String,
+    /// The source of this error.
+    source: core::num::ParseIntError,
+
+    /// For displaying - see Display impl with nice error message below
     bits: u8,
-    // We could represent this as a single bit but it wouldn't actually derease the cost of moving
-    // the struct because String contains pointers so there will be padding of bits at least
-    // pointer_size - 1 bytes: min 1B in practice.
+
+    /// We could represent this as a single bit but it wouldn't actually derease the cost of moving
+    /// the struct because String contains pointers so there will be padding of bits at least
+    /// pointer_size - 1 bytes: min 1B in practice.
     is_signed: bool,
-    pub(crate) source: core::num::ParseIntError,
 }
 
 impl ParseIntError {
     /// Returns the input that was attempted to be parsed.
     pub fn input(&self) -> &str { &self.input }
+
+    /// Returns the `num::ParseIntError` encountered while parsing input.
+    pub fn source(&self) -> &core::num::ParseIntError { &self.source }
+
+    /// Returns the input and source error.
+    pub fn into_input_source(self) -> (String, core::num::ParseIntError) {
```

Comment:
This should require `alloc` since we'll have `InputString` there eventually.

Another issue is the naming - it sounds like "input source" not "input and source". The canonical name for these in Rust is `into_parts`.

Reply:
The whole module requires "alloc". 

`into_parts` is good, will use.